app/controllers/users_controller.rb

```diff
@@ -17,6 +17,11 @@ def create
   end
 
   def show
+    if current_user
+      @user = current_user
+    else
+      render file: 'public/404'
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.